test/test_poisson_solvers.jl

```diff
@@ -172,44 +171,86 @@ end
 PB = (Periodic, Bounded)
 topos = collect(Iterators.product(PB, PB, PB))[:]
 
+two_dimensional_topologies = [
+                              (Flat,     Bounded,  Bounded),
+                              (Bounded,  Flat,     Bounded),
+                              (Bounded,  Bounded,  Flat),
+                              (Flat,     Periodic, Bounded),
+                              (Periodic, Flat,     Bounded),
+                              (Periodic, Bounded,  Flat),
+                             ]
+
 @testset "Poisson solvers" begin
     @info "Testing Poisson solvers..."
 
     for arch in archs
         @testset "Poisson solver instantiation [$(typeof(arch))]" begin
             @info "  Testing Poisson solver instantiation [$(typeof(arch))]..."
             for FT in float_types
-                @test poisson_solver_instantiates(arch, FT, 32, 32, 32, FFTW.ESTIMATE)
-                @test poisson_solver_instantiates(arch, FT, 1,  32, 32, FFTW.MEASURE)
-                @test poisson_solver_instantiates(arch, FT, 32,  1, 32, FFTW.ESTIMATE)
-                @test poisson_solver_instantiates(arch, FT,  1,  1, 32, FFTW.MEASURE)
+
+                grids_3d = [
+                            RegularRectilinearGrid(FT, size=(2, 2, 2), extent=(1, 1, 1)),
+                            RegularRectilinearGrid(FT, size=(1, 2, 2), extent=(1, 1, 1)),
+                            RegularRectilinearGrid(FT, size=(2, 1, 2), extent=(1, 1, 1)),
+                            RegularRectilinearGrid(FT, size=(2, 2, 1), extent=(1, 1, 1))
+                           ]
+
+                grids_2d = [RegularRectilinearGrid(FT, size=(2, 2), extent=(1, 1), topology=topo)
+                            for topo in two_dimensional_topologies]
+
+
```

Comment:
Any reason why we don't test 1d topologies?

Reply:
Pressure solve is trivial in 1D? Tests are expensive so I don't like to add more tests than we strictly need.

Reply:
Okay, no problem.